realm/src/main/java/io/realm/RealmList.java

```diff
@@ -94,16 +155,24 @@ public void clear() {
      */
     @Override
     public E remove(int location) {
-        view.remove(location);
-        return null;
+        if (managedMode) {
+            view.remove(location);
+            return null; // TODO Return the proper element not null
```

Comment:
I would say document it for now and implement it later, as the original behaviour was wrong. Depending on how fast we release, we might be able to add it though